### ISSUE TITLE
REST API root "/" not handled properly

### ISSUE THREAD

**Issue by crandmck** [strongloop/loopback-explorer#48]
If you set ` "restApiRoot": "/",` in `config.json` Explorer does not handle it properly.

http://screencast.com/t/Z4pDmegA3

Explorer tries to go to `http:// localhost:3000//books`

**Comment by marcopeise** [comment]
same problem here. got a workaround within loopback-boot/lib/executor.js:
```javascript
119 if(restApiRoot=='/'){
120     restApiRoot = ''
121   }else{
122     assert(/^\//.test(restApiRoot),     'app.restApiRoot must start with "/"')
123   }
```

**Comment by bajtos** [comment]
@marcopeise would you mind submitting a pull request to fix the problem?

It may be enough to use `urlJoin` instead of string concatenation on [lib/swagger.js#L175](https://github.com/strongloop/loopback-explorer/blob/9a9b7899a08557ea76cef091070ad73a9c90ad9f/lib/swagger.js#L175).

**Comment by marcopeise** [comment]
@bajtos well, host + initialPath is still valid when used urlJoin. (i.e.: "http://localhost:3000/")
couldn't find the right code where basePath and routePath where joined. but submitted the workaround as pull request.

**Comment by bajtos** [comment]
To be honest, this looks like a swagger UI bug to me. LoopBack returns the following fields:

```js
// GET /explorer/resources/products

{ 
  apiVersion: '1.6.3',
  swaggerVersion: '1.2',
  basePath: 'http://127.0.0.1:62232/',
  resourcePath: '/resources',
  apis: [],
  consumes:
   [ 'application/json',
     'application/x-www-form-urlencoded',
     'application/xml',
     'text/xml' ],
  produces:
   [ 'application/json',
     'application/xml',
     'text/xml',
     'application/javascript',
     'text/javascript' ],
  models: { product: { id: 'product', properties: [Object], required: [] } }
}
```

It is the Swagger UI that concatenates `basePath: 'http://127.0.0.1:62232/'` with  `resourcePath: '/resources'` to produce the final value `http://127.0.0.1:62232//resources`.

A workaround for loopback-explorer is to modify the code building `basePath` to remove any trailing slashes.

**Comment by bajtos** [comment]
Hang on, the conclusion I made above is not correct. I need to investigate this little bit more.

**Comment by bajtos** [comment]
Actually I was correct, I have amended my comment.

**Comment by bajtos** [comment]
The swagger issue: https://github.com/swagger-api/swagger-js/issues/201

**Comment by bajtos** [comment]
The Swagger UI issue: https://github.com/swagger-api/swagger-ui/issues/819

**Comment by bajtos** [comment]
The fix was released as `loopback-explorer@1.6.4`
